### PR TITLE
Optimize and inline bitboard.Grow

### PR DESCRIPTION
**Pull request by zenazn**
This results in the following microbenchmarks on my Early 2013 2.7 GHz
Mac Book Pro using Go 1.6.2:

```
name            old time/op    new time/op    delta
EmptyHasRoad-8    20.7ns ± 5%    20.4ns ± 4%     ~            (p=0.354 n=10+9)
FullHasRoad-8      197ns ± 7%     162ns ± 6%  -18.04%        (p=0.000 n=10+10)
HasRoadWindy-8    59.4ns ± 4%    53.4ns ± 6%  -10.11%        (p=0.000 n=10+10)
```

### CODE REVIEW

File: bitboard/bits.go [hunk 1]
```diff
@@ -41,12 +46,7 @@ func Flood(c *Constants, within uint64, seed uint64) uint64 {
 }
 
 func Grow(c *Constants, within uint64, seed uint64) uint64 {
-	next := seed
-	next |= (seed << 1) &^ c.R
```
Comment (zenazn) [inline bitboard/bits.go]: The important part here is that, since we're using `c.Mask` below, we don't also need to mask using `c.R`

**Comment by zenazn** [comment]
And apparently I don't understand what `c.L` and `c.R` are. Oops.

**Comment by zenazn** [comment]
Ah, bingo: this doesn't work on 8x8 boards, and the diagram I drew myself was smaller than that

**Comment by nelhage** [comment]
I think it also doesn't work on smaller boards; `L` and `R` here prevent wraparound when shifting squares at the edge of the board.

**Comment by zenazn** [comment]
I also misinterpreted the bitboard as embedding a NxN board in the top corner of an 8x8, which is also not what you're doing. So many things were wrong with my mental model of the universe :)

**Pull request closed without merging**